test_stale_repositories.go

```diff
@@ -0,0 +1,233 @@
+		return
+	}
+	newIssue := &issue{
+		Title: issueTitle,
+		Body:  body,
+	}
+	buf := new(bytes.Buffer)
+	json.NewEncoder(buf).Encode(newIssue)
+	req, err := http.NewRequest("POST", githubPOSTISSUES, buf)
+	if err != nil {
+		log.Print("Failed at CreateIssue")
+		return
+	}
+	oauthClient.Do(req)
+}
+func getAllFlaggedRepositories(oauthClient *http.Client, flaggedRepositories *map[string]bool) error {
+	req, err := http.NewRequest("GET", awesomeGoGETISSUES, nil)
+	if err != nil {
+		log.Print("Failed to get all issues")
+		return err
+	}
+	res, err := oauthClient.Do(req)
+	if err != nil {
+		log.Print("Failed to get all issues")
+		return err
+	}
+	target := []issue{}
+	defer res.Body.Close()
+	json.NewDecoder(res.Body).Decode(&target)
+	for _, i := range target {
+		if i.Title == issueTitle {
+			repos := getRepositoriesFromBody(i.Body)
+			fmt.Println(repos)
+			for _, repo := range repos {
+				(*flaggedRepositories)[repo] = true
+			}
+		}
+	}
+	return nil
+}
+func containsOpenIssue(link string, openIssues map[string]bool) bool {
+	_, ok := openIssues[link]
+	if ok {
+		return true
+	}
+	return false
+}
+func testCommitAge(href string, oauthClient *http.Client, staleRepos *[]string, openIssues map[string]bool) {
+	var isValidRepo bool
+	var isAged bool
+	var ownerRepo string
+	isValidRepo = reGithubRepo.MatchString(href)
+	var apiCall string
+	var respObj []map[string]interface{}
+	now := time.Now()
+	since := now.Add(-1 * 365 * 24 * numberOfYears * time.Hour)
+	sinceQuery := since.Format(time.RFC3339)
+	if isValidRepo {
+		issueExists := containsOpenIssue(href, openIssues)
+		if issueExists {
+			log.Printf("issue already exists for %s\n", href)
+			return
+		}
+		ownerRepo = strings.ReplaceAll(href, "https://github.com", "")
+		apiCall = fmt.Sprintf(githubGETCOMMITS, ownerRepo)
+		req, err := http.NewRequest("GET", apiCall, nil)
+		if err != nil {
+			log.Printf("Failed at repository %s\n", href)
+			return
+		}
+		q := req.URL.Query()
+		q.Add("since", sinceQuery)
+		req.URL.RawQuery = q.Encode()
+		resp, err := oauthClient.Do(req)
+		if err != nil {
+			log.Printf("Failed at repository %s\n", href)
+			return
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == 301 {
+			*staleRepos = append(*staleRepos, href+movedPermanently)
+			log.Printf("%s returned 301", href)
+		} else if resp.StatusCode == 302 {
+			*staleRepos = append(*staleRepos, href+status302)
+			log.Printf("%s returned 302", href)
+		} else if resp.StatusCode >= 400 {
+			*staleRepos = append(*staleRepos, href+deadLinkMessage)
+			log.Printf("%s might not exist!", href)
+		} else {
+			json.NewDecoder(resp.Body).Decode(&respObj)
+			isAged = len(respObj) == 0
+		}
+		if isAged {
+			log.Printf("%s has not had a commit in a while", href)
+			*staleRepos = append(*staleRepos, href)
+			ctr++
+		}
+	}
+}
+func testStaleRepository() {
+	query := startQuery()
+	var staleRepos []string
+	addressedRepositories := make(map[string]bool)
+	oauth := os.Getenv("GITHUB_OAUTH_TOKEN")
```

Comment:
as variable environment setting in github workflow `GITHUB_OAUTH_TOKEN`

Reply:
added environment variable OAUTH_TOKEN. @avelino, you need to create a new OAUTH token and create a secret for this repository with the same name